backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointService.java

```diff
@@ -180,17 +200,58 @@ public Endpoint createEndpoint(@Context SecurityContext sec, @NotNull @Valid End
                 CamelProperties properties = endpoint.getProperties(CamelProperties.class);
                 String processorName = "p-" + endpoint.getAccountId() + "-" + UUID.randomUUID();
                 properties.getExtras().put(OB_PROCESSOR_NAME, processorName);
-                String processorId = setupOpenBridgeProcessor(endpoint, properties, processorName);
-
-                // TODO find a better place for these, that should not be
-                //       visible to users / OB actions
-                //       See also CamelTypeProcessor#callOpenBridge
-                properties.getExtras().put(OB_PROCESSOR_ID, processorId);
+                try {
+                    processorId = setupOpenBridgeProcessor(endpoint, properties, processorName);
+                    endpoint.setStatus(EndpointStatus.PROVISIONING);
+
+                    // TODO find a better place for these, that should not be
+                    //       visible to users / OB actions
+                    //       See also CamelTypeProcessor#callOpenBridge
+                    properties.getExtras().put(OB_PROCESSOR_ID, processorId);
+                    needChecking = true;
+                } catch (WebApplicationException we) {
+                    we.printStackTrace();
+                    endpoint.setStatus(EndpointStatus.FAILED);
+                }
 
             }
         }
 
-        return resources.createEndpoint(endpoint);
+        Endpoint createdEndpoint = resources.createEndpoint(endpoint);
+        if (obEnabled && needChecking && createdEndpoint.getSubType().equals("slack")) {
+            triggerReadyCheck(createdEndpoint.getId(), processorId);
+        }
+        return createdEndpoint;
+    }
+
+
+    private void triggerReadyCheck(UUID endpointId, String processorId) {
+
+        JobDataMap jdm = new JobDataMap();
+        jdm.put("epid", endpointId);
+        jdm.put("pid", processorId);
+        JobDetail jd = JobBuilder.newJob().ofType(ReadyCheck.class)
+                .withIdentity("myJob")
+                .usingJobData(jdm)
+                .build();
+
+        Trigger trigger = TriggerBuilder.newTrigger()
+                .withIdentity(new TriggerKey("myJobs", "obWorker"))
+                .startNow()
+                .withSchedule(
+                        SimpleScheduleBuilder.simpleSchedule()
+                                .repeatForever()
+                                .withIntervalInSeconds(10)
+                )
+                .build();
+
+        try {
+            quartz.scheduleJob(jd, trigger);
+        } catch (SchedulerException e) {
+            em.find(Endpoint.class, endpointId).setStatus(EndpointStatus.FAILED);
+            e.printStackTrace();
+        }
```

Comment:
I think we should probably move this code to other service. What worries me here is that it has a "repeatForever" - Maybe we should have some kind of timeout or check it less often?

Reply:
Yes, that could probably go into `BridgeHelper`.

Reply:
The code changed. Comment no longer relevant.